packages/expo-modules-core/ios/Core/Views/SwiftUI/SwiftUIViewDefinition.swift

```diff
@@ -67,38 +67,88 @@ extension ExpoSwiftUI {
   /**
    A definition representing the native SwiftUI view to export to React.
    */
-  public final class ViewDefinition<Props: ViewProps, ViewType: View<Props>>: ExpoModulesCore.ViewDefinition<HostingView<Props, ViewType>> {
+  public final class ViewDefinition<Props: ViewProps, ViewType: View<Props>>: ObjectDefinition, AnyViewDefinition {
```

Comment:
remove the inheritance from `ExpoModulesCore.ViewDefinition` and copy necessary code here. because we have more different implementations than `ExpoModulesCore.ViewDefinition`

Reply:
Could we maybe align these implementations and make our ViewDefinition return AppleViews as well? Don't think I get why we need to duplicate those definitions

Reply:
good call! the main point is the logic are quite different between UIView and SwiftUI views, specifically for createView. i've tried to rollback to use inheritance here. at the same time i have to remove the UIView constraint for  the base ViewDefinition. it' basically at https://github.com/expo/expo/pull/35553/files#diff-773bcb83edb82f60778547d35ebbe6a7d0b8524934c07c1b1452f6efe3d8fe32L6